eda/src/autogluon/eda/analysis/shift.py

```diff
@@ -0,0 +1,256 @@
+        The label internally used for the classifier 2 sample test, the only reason to change it is in the off chance
+        that the default value is a column in the data.
+    classifier_kwargs : dict, default = {}
+        The kwargs passed to the classifier, a member of classifier_class
+    num_permutations: int, default = 1000
+        The number of permutations used for any permutation based method
+
+    State attributes
+    ----------------
+    state.xshift_results: outputs the results of XShift detection,
+        dict of
+            - 'detection_status': bool, True if detected
+            - 'test_statistic': float, the C2ST statistic
+            - 'pvalue': float, the p-value using permutation test
+            - 'pvalue_threshold': float, the decision p-value threshold
+            - 'feature_importance': DataFrame, the feature importance dataframe, if computed
+    """
+
+    def __init__(self,
+                 classifier_class: Any = TabularPredictor,
+                 compute_fi: bool = True,
+                 pvalue_thresh: float = 0.01,
+                 eval_metric: str = 'roc_auc',
+                 sample_label: str = 'i2vkyc0p64',
+                 classifier_kwargs: dict = None,
+                 num_permutations: int = 1000,
+                 parent: Union[None,AbstractAnalysis] = None,
+                 children: List[AbstractAnalysis] = [],
+                 **kwargs) -> None:
+        super().__init__(parent, children, **kwargs)
+        if classifier_kwargs is None:
+            classifier_kwargs = {}
+        self.classifier_kwargs = classifier_kwargs
+        self.classifier_class = classifier_class
+        self.compute_fi = compute_fi
+        named_metrics = BINARY_METRICS
+        assert eval_metric in named_metrics.keys(), \
+            'eval_metric must be one of [' + ', '.join(named_metrics.keys()) + ']'
+        self.eval_metric = named_metrics[eval_metric]
+        self.C2ST = Classifier2ST(classifier_class,
+                                  sample_label=sample_label,
+                                  eval_metric=self.eval_metric,
+                                  compute_fi = compute_fi,
+                                  classifier_kwargs=classifier_kwargs)
+        self.fi_scores = None
+        self.compute_fi = compute_fi
+        self.pvalue_thresh = pvalue_thresh
+        self.num_permutations = num_permutations
+
+    def can_handle(self, state: AnalysisState, args: AnalysisState) -> bool:
+        return self.all_keys_must_be_present(args, ['train_data', 'test_data'])
+
+    def _fit(self, state: AnalysisState, args: AnalysisState, **fit_kwargs):
+        """ Fit method.  `args` can contain
+        - 'train_data': pd.DataFrame, required
+        - 'test_data': pd.DataFrame, required
+        - 'label': str, optional
+            The Y variable that is to be predicted (if it appears in the train/test data then it will be removed)
+        """
+        self.can_handle(state, args)
```

Comment:
This is not needed - `fit()` all will do this automatically and if the result is `True`, it'll pass control over to `_fit()`,